pkgs/by-name/lu/lunarvim/package.nix

```diff
@@ -0,0 +1,121 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, makeWrapper
+, cargo
+, curl
+, fd
+, fzf
+, git
+, gnumake
+, gnused
+, gnutar
+, gzip
+, neovim
+, nodejs
+, nodePackages
+, ripgrep
+, tree-sitter
+, unzip
+, nvimAlias ? false
+, viAlias ? false
+, vimAlias ? false
+, globalConfig ? ""
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "lunarvim";
+  version = "1.3.0";
+
+  src = fetchFromGitHub {
+    owner = "LunarVim";
+    repo = "LunarVim";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-z1Cw3wGpFDmlrAIy7rrjlMtzcW7a6HWSjI+asEDcGPA=";
+  };
+
+  nativeBuildInputs = [
+    gnused
+    makeWrapper
+  ];
+
+  buildInputs = [
+    cargo
+    curl
+    fd
+    fzf
+    git
+    gnumake
+    gnutar
+    gzip
+    neovim
+    nodejs
+    nodePackages.neovim
+    ripgrep
+    tree-sitter
+    unzip
+  ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir $out
+    cp init.lua utils/installer/config.example.lua $out
+    cp -r lua snapshots $out
+
+    mkdir $out/bin
+    cp utils/bin/lvim.template $out/bin/lvim
+    chmod +x $out/bin/lvim
+
+    # LunarVim automatically copies config.example.lua, but we need to make it writable.
+    sed -i "2 i\\
+            if [ ! -f \$HOME/.config/lvim/config.lua ]; then \\
+              cp $out/config.example.lua \$HOME/.config/lvim/config.lua \\
+              chmod +w \$HOME/.config/lvim/config.lua \\
+            fi
+    " $out/bin/lvim
+
+    substituteInPlace $out/bin/lvim \
+      --replace NVIM_APPNAME_VAR lvim \
+      --replace RUNTIME_DIR_VAR \$HOME/.local/share/lvim \
+      --replace CONFIG_DIR_VAR \$HOME/.config/lvim \
+      --replace CACHE_DIR_VAR \$HOME/.cache/lvim \
+      --replace BASE_DIR_VAR $out \
+      --replace nvim ${neovim}/bin/nvim
+
+    echo ${ lib.strings.escapeShellArg globalConfig } > $out/global.lua
```

Comment:
It is equivalent to install files on the root dir. 

Why?

Reply:
Whoops, my bad.